src/recipe/carne_asade.rs

```diff
@@ -0,0 +1,384 @@
+
+        let header: Header = Header::parse(&header_buffer);
+        let metadata = metadata::Metadata {
+            size: header.size.clone(),
+            date_of_recipe: header.date_of_recipe.clone(),
+            time_of_recipe: header.time_of_recipe.clone(),
+            number_of_steps: header.number_of_steps.clone(),
+            steps: header
+                .steps
+                .map(|i| String::from(STEPS_BY_NAME[i]))
+                .to_vec(),
+            units: 1.0,
+            granularity: header.granularity.clone(),
+        };
+        metadata.store(dir.join("header.json"));
+
+        let step_count = u32::from(header.number_of_steps);
+        let total_bytes = header.size * step_count * 2;
+        let chunks =
+            CarneAsada::calculate_chunks(total_bytes, MAX_BYTES, MAX_THREADS, step_count, DTYPE);
+
+        let mut generated_files = Vec::<(usize, u32, String)>::new();
+        for chunk in chunks {
+            for thread in chunk {
+                generated_files.append(&mut CarneAsadaGaucamole::parse_guacamole(
+                    &dir,
+                    &mut reader,
+                    thread.0,
+                    thread.1,
+                    522,
+                    header.number_of_steps,
+                    &header.unit_conversion,
+                ));
+            }
+        }
+        generated_files.sort_by(|a, b| match a.0.cmp(&b.0) {
+            Ordering::Equal => a.1.cmp(&b.1),
+            other => other,
+        });
+
+        CarneAsadeFile::merge(&dir, &generated_files);
+
+        Some(Event {
+            event_type: 0,
+            payload: RecipeParsed {
+                name: todo!(),
+                guacamole: todo!(),
+                beans: todo!(),
+                lettuce: todo!(),
+            },
+        })
+    }
+
+    fn identifier(&self) -> String {
+        String::from("carne_asada")
+    }
+}
+
+pub struct CarneAsadeFile {}
+
+impl CarneAsadeFile {
+    pub fn read_chunk(
+        reader: &mut BufReader<File>,
+        onset: u32,
+        offset: u32,
+        start: u32,
+    ) -> Vec<u8> {
+        let mut buffer = vec![0; usize::try_from(offset - onset).unwrap()];
+        reader
+            .seek(SeekFrom::Start((start + onset).into()))
+            .unwrap();
+        reader.read_exact(&mut buffer).expect("Could not read.");
+        buffer
+    }
+
+    pub fn write_chunk(
+        dir: &PathBuf,
+        onset: u32,
+        offset: u32,
+        step: usize,
+        guac: &Vec<f64>,
+    ) -> (usize, u32, String) {
+        let filename = format!("{}_{}_{}.json", onset, offset, step);
+        let file = File::create(dir.join(&filename)).expect("Could not create file.");
+        let writer = BufWriter::new(file);
+        serde_json::to_writer(writer, &guac).expect("Could not write json.");
+        (step, onset, filename)
+    }
+
+    pub fn merge(dir: &PathBuf, files: &Vec<(usize, u32, String)>) {
```

Comment:
@hwadii 😅 oh you know.. normal json stuff

Reply:
haha this is unhinged ngl

Reply:
it's a private api 😉 no need for concern